include/experimental/__p1673_bits/blas1_dot.hpp

```diff
@@ -100,10 +101,18 @@ Scalar dot(
                 v1.static_extent(0) == v2.static_extent(0));
 
   using size_type = std::common_type_t<SizeType1, SizeType2>;
-  for (size_type k = 0; k < v1.extent(0); ++k) {
-    init += v1(k) * v2(k);
-  }
-  return init;
+  using scalar_type = std::common_type_t<ElementType1, ElementType2, Scalar>;
+  using std::ranges::iota_view;
+  using std::ranges::begin;
+  using std::ranges::end;
+
+  iota_view range{size_type{}, v1.extent(0)};
+
+  Scalar sum = std::transform_reduce(exec, begin(range), end(range), init,
+    std::plus<scalar_type> {},
```

Comment:
Would you consider using the `std::plus<void>` specialization, which deduces argument and return types?  That would fix the `scalar_type` issue above.  It lets the two input types differ, and thus preserves the semantics of the original sequential code.

https://en.cppreference.com/w/cpp/utility/functional/plus_void

```suggestion
    std::plus<void>{},
```